### ISSUE TITLE
Bug with `eval` on automata with two inputs over `{0, 1} {0, 1}`

### ISSUE THREAD

**Issue by firetto** [firetto/Walnut#8]
The following creates a kind of inexplicable error.

```
reg foo {0,1} {0,1} "([1,0])*[0,1][0,0]*";

eval tmp "?lsd_2 $foo(x,x)":

[Walnut]$
java.lang.NullPointerException
     at Token.Function.act(Function.java:79)
     at Main.Computer.compute(Computer.java:111)
     at Main.Computer.<init>(Computer.java:49)
     at Main.Prover.eval_def_commands(Prover.java:466)
     at Main.Prover.dispatch(Prover.java:328)
     at Main.Prover.readBuffer(Prover.java:280)
     at Main.Prover.run(Prover.java:234)
     at Main.Prover.main(Prover.java:201)
null
     : char at 8
     : eval tmp "?lsd_2 $foo(x,x)":
```

whereas

```
reg foo2 lsd_2 lsd_2 "([1,0])*[0,1][0,0]*":
eval tmp "?lsd_2 $foo2(x,x)":
```

does not give the error.

**Comment by firetto** [comment]
Automaton generated by `foo`:
```
{0, 1} {0, 1} 

0 0
1 0 -> 0
0 1 -> 1

1 1
0 0 -> 1
```

Automaton generated by `foo2`:
```
lsd_2 lsd_2

0 0
1 0 -> 0
0 1 -> 1

1 1
0 0 -> 1
```

**Comment by jn1z** [comment]
I don't know the underlying solution, but the problem seems due to the Automaton's NumberSystem not being fully initialized (edit: or Function.java assuming that it is).

The hidden exception is:
> java.lang.NullPointerException: Cannot read field "equality" because the return value of "java.util.List.get(int)" is null

This is because A.NS.get(i) returns a null, in line 79 of Function.java. In particular, A.NS is [null, null] at this point.

**Comment by firetto** [comment]
This bug was fixed in a0a4db3103cdb981309f9036e207717ae283f1f9, I forgot to close this issue.
